props/hev.h

```diff
@@ -0,0 +1,480 @@
+           ((now >= next_tick_) ? (next_tick_ = now + tick_interval, true) : false);
+  }
+
+  // Returns true if timer is active and interval has elapsed
+  bool ready(uint32_t interval) {
+    return active_ && ((millis() - start_) >= interval);
+  }
+};
+
+class Hev : public PROP_INHERIT_PREFIX PropBase {
+public:
+  Hev() : PropBase() {}
+  const char* name() override { return "Hev"; }
+
+  int health_ = 100;
+  int armor_ = 100;
+
+  HEVTimer clash_timer_;
+  HEVTimer random_event_timer_;
+  HEVTimer post_death_cooldown_timer_;
+  HEVTimer hazard_delay_timer_;
+  HEVTimer hazard_damage_timer_;
+  HEVTimer health_increase_timer_;
+  HEVTimer armor_increase_timer_;
+
+  enum Hazard {
+    HAZARD_NONE = 0,
+    HAZARD_BIO = 1,
+    HAZARD_RAD = 2,
+    HAZARD_BLO = 3,
+    HAZARD_CHE = 4,
+    HAZARD_HEA = 5,
+    HAZARD_SHO = 6,
+  };
+
+  Hazard current_hazard_ = HAZARD_NONE;
+
+  // Calculate Physical Damage
+  void DoDamage(int damage, bool quiet = false) {
+    int tens = health_ / 10;
+    if (armor_ >= damage) {
+      armor_ -= ceilf(damage * 0.80 / 2);
+      health_ -= ceilf(damage * 0.20);
+    } else if (armor_ > 0) {
+      int excess_physical = damage - armor_;
+      health_ -= excess_physical;
+      // Make armor compromised sound play before setting armor to 0
+      SaberBase::DoEffect(EFFECT_USER2, 0.0, 0);
+      PVLOG_NORMAL << "Armor Compromised!\n";
+      armor_ = 0;
+    } else {
+      health_ -= damage;
+    }
+
+    // Enforce minimum values
+    if (health_ < 0) health_ = 0;
+    if (armor_ < 0) armor_ = 0;
+
+    // Damage
+    if (!quiet) SaberBase::DoEffect(EFFECT_STUN, 0.0, damage);
+    
+    // Death Sound
+    if (health_ == 0) {
+      SaberBase::DoEffect(EFFECT_EMPTY, 0.0, 100);
+      return;
+    }
+    
+    // Health Alert - only plays when health enters a new multiple of 10
+    int new_tens = health_ / 10;
+    if (tens != new_tens) {
+      SaberBase::DoEffect(EFFECT_USER1, 0.0, 0);
+    }
+
+    // Print Damage, Health and Armor
+    PVLOG_NORMAL << "DAMAGE: -" << damage << "\n";
+    PVLOG_NORMAL << "HEALTH: " << health_ << " / ";
+    PVLOG_NORMAL << "ARMOR: " << armor_ << "\n";
+  }
+
+  // Armor Readout
+  void armor_readout() {
+    PVLOG_NORMAL << "Current Armor: " << armor_ << "\n";
+    
+    // Play random "fuzz" sound only if armor is above 0
+    if (armor_ > 0) {
+      if (random(2) == 0) {
+          hybrid_font.PlayCommon(&SFX_fuzz_high);
+      } else {
+          hybrid_font.PlayCommon(&SFX_fuzz_low);
+      }
+
+      // Small delay to ensure fuzz finishes before armor readout plays.
+      delay(475);
+    }
+
+    // Play armor value
+    SFX_armor.SelectFloat(armor_ / 100.0);
+    hybrid_font.PlayCommon(&SFX_armor);
+  }
+
+  // Clashes
+  void Clash(bool stab, float strength) override {
+    // Don't allow Clashes if dead.
+    if (health_ == 0) {
+      return;
+    }
+
+    // If HEVTimer and PropBase's clash_timeout_ are true to activate Clash.
```

Comment:
This comment doesn't really make any sense to me.
Also, comments are supposed to describe *why* things are the way they are, not just what the code is doing.